lorawan-encoding/src/parser.rs

```diff
@@ -1059,12 +1060,13 @@ impl<'a> FHDR<'a> {
         (u16::from(self.0[6]) << 8) | u16::from(self.0[5])
     }
 
-    /// Gives the piggy-backed MAC ommands associated with the given payload.
-    pub fn fopts(&self) -> MacCommandIterator {
-        let f_opts_len = FCtrl(self.0[4], self.1).f_opts_len();
-        let start = 7;
-        let end = 7 + f_opts_len as usize;
-        parse_mac_commands(&self.0[start..end], self.1)
+    /// Gives the size of FOpts.
+    pub fn fopts_len(&self) -> u8 {
+        FCtrl(self.0[4], self.1).f_opts_len()
+    }
+
+    pub fn data(&self) -> &[u8] {
```

Comment:
I am thinking - if I read just the function name, I will not be sure what this returns, so maybe a short comment with that will be helpful.

Reply:
Right you are, I did some cleaning here. I also removed the fopts_len fn as it duplicates what already happens in FCtrl